src/components/course/data/utils.jsx

```diff
@@ -138,16 +138,35 @@ export function getActiveCourseRun(course) {
  * @param {object} course
  * @returns List of course runs.
  */
-export function getAvailableCourseRuns(course) {
+export function getAvailableCourseRuns({ course, isEnrollableBufferDays }) {
   if (!course?.courseRuns) {
     return [];
   }
-  return course.courseRuns
-    .filter((courseRun) => (
-      courseRun.isMarketable
-      && courseRun.isEnrollable
-      && !isArchived(courseRun)
-    ));
+  const availableCourseRunsFilter = (courseRun) => {
+    if (!courseRun.isMarketable || isArchived(courseRun)) {
+      return false;
+    }
+
+    if (isEnrollableBufferDays === undefined) {
+      return courseRun.isEnrollable;
+    }
+
+    const today = new Date();
+    if (courseRun.enrollmentStart && today < courseRun.enrollmentStart) {
```

Comment:
[clarification/suggestion] What is the value of `courseRun.enrollmentStart` and `courseRun.enrollmentEnd`? Are these the raw datetime strings from the API or are they casted as `Date` objects (or `dayjs`)? If not, you may need to do so.

It might be good to use `dayjs` to help simplify some of the date logic?

```jsx
const today = dayjs();
if (courseRun.enrollmentStart && today.isBefore(dayjs(courseRun.enrollmentStart)) { ... }

// ...

const bufferedEnrollDeadline = dayjs(courseRun.enrollmentEnd).add(isEnrollableBufferDays, 'day');
return today.isBefore(bufferedEnrollDeadline);
```